client/components/app.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PartySize from './partySize';
+import Calendar from './calendar';
+import Time from './time';
+import Availability from './availability';
+import TimesBookedToday from './timesBookedToday';
+import TimeslotsLeft from './timeslotsLeft';
+import SaveThisRestaurant from './saveThisRestaurant';
+import { removeHyphen, mapTo24Hr, numStrToPrimitive, Hr24ToAMPM, strippedDateTime } from './helper';
+import styles from '../styles/styles.css';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dateTime: '2020-03-27 9:00 PM',
+      partySize: '2', // user-supplied
+      stage: 'findTable',
+      displayedSlots: [],
+      timesBookedToday: null,
+    };
+
+    // the user will have supplied some values from the home page before the restaurant page
+    this.restId = 1;
+    this.restaurantName = 'Alinea';
+
+    this.handleChangeParty = this.handleChangeParty.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeTime = this.handleChangeTime.bind(this);
+    this.handleFindTable = this.handleFindTable.bind(this);
+  }
+
+  componentDidMount() {
+    // fetch number of times booked today
+    fetch(`/reservations/timesBookedToday/${this.restId}`)
+      .then(res => res.json())
+      .then((response) => {
+        if (response !== '-1') this.setState({ timesBookedToday: response });
+      });
+  }
+
+  handleChangeParty(e) {
+    this.setState({
+      partySize: e.target.value,
+      stage: 'findTable',
+    });
+  }
+
+  handleChangeDate(e) {
+    const { dateTime } = this.state;
```

Comment:
"handleDateSelection"

also, it looks like you are not using the 2 const variables defined within it.